lib/src/utils.dart

```diff
@@ -673,7 +674,8 @@ Future<T> retry<T>(
   double randomizationFactor = 0.25,
   Duration maxDelay = const Duration(seconds: 30),
   int maxAttempts = 8,
-  FutureOr<bool> Function(Exception)? retryIf,
+  Exception Function(Exception, StackTrace)? mapException,
+  FutureOr<bool> Function(Exception, StackTrace)? retryIf,
```

Comment:
Is there ever a good reason make the decision on whether or not to retry based on `StackTrace`?

Reply:
Hah, probably not.